docs/proposals/cross-mesh-ux.md

```diff
@@ -0,0 +1,144 @@
+    hostname: foo.example.com
+    tags:
+      port: "8080"
+---
+type: MeshGatewayRoute
+mesh: expose
+name: edge-gateway-route
+selectors:
+  - match:
+      kuma.io/service: edge-gateway
+conf:
+  http:
+    rules:
+      - matches:
+          - path:
+              match: PREFIX
+              value: /api
+        backends:
+          - destination:
+              kuma.io/service: server
+```
+
+</details>
+
+### `expose` UX
+
+The main UX feature will be a mechanism to configure a
+`MeshGateway` to permit traffic from services in one
+`Mesh` to reach the services behind the `MeshGateway`.
+The configuration will make it possible to expose `MeshGatewayRoute`s depending
+on cross-mesh traffic's source `Mesh` and source _tags_.
+
+First of all, we require users to explicitly enable cross-mesh traffic on
+the `MeshGateway` listener by marking it `internal` as well as mTLS on
+any meshes involved in cross-mesh traffic.
+This option will allow us to configure the corresponding Envoy listener
+with all mesh CAs.
+
+```
+type: MeshGateway
+mesh: expose
+name: edge-gateway
+selectors:
+- match:
+    kuma.io/service: edge-gateway
+conf:
+  listeners:
+  - port: 8090
+    protocol: HTTPS
+    hostname: expose.mesh # optional in general
+    internal: true
+    tags:
+      name: mesh-listener
+```
+
+#### Permissions
+
+We will take advantage of the new `MeshTrafficPermission` to define which
+cross-mesh traffic is allowed for a matching `MeshGateway` listener and
+`routeName`. It's being discussed in https://github.com/kumahq/kuma/issues/4222
+
+For the first iteration, traffic will be permitted from all services in all
+meshes.
+
+#### Inline `MeshGatewayRoute`
+
+We decided against adding cross-mesh access control to the `MeshGatewayRoute`
+resource, as another type of `conf.http.rules.matches` entry or
+potentially at the top level under `conf.http`.
+
+It reduces our flexibility in the future.
+
+### `consume` UX
+
+In order for `consume` services to communicate with `expose`, we need to give
+those services a way to contact the `MeshGateway`. A `MeshGateway` doesn't have
+an address or concrete identity beyond the identities of the `kuma-dp` proxies that
+serve the gateway.
+
+#### Virtual outbounds
+
+Every `MeshGateway` listener can be given a `hostname`. For cross-mesh, we will
+require one be set and use it to generate a virtual outbound.
+
+The listener from above
+
+```
+type: MeshGateway
+mesh: expose
+name: edge-gateway
+selectors:
+- match:
+    kuma.io/service: edge-gateway
+conf:
+  listeners:
+  - port: 8090
+    protocol: HTTPS
+    hostname: expose.mesh
+    tags:
+      name: mesh-listener
+```
+
+should give us the DNS hostname `expose.mesh` as well as a virtual outbound
+like:
+
+```
+address: 240.0.0.1
+port: 8090
+tags:
+  kuma.io/mesh: expose # we need to keep track of the mesh somehow
+  kuma.io/service: edge-gateway
+  name: mesh-listener
+```
+
+giving access from `consume` to `expose` at `http://expose.mesh:8090/api`.
```

Comment:
It's not exactly clear whether this vob is auto-generated or not.

Reply:
yes it is, added to the proposal